docs/manual/manual_compiler.html

```diff
@@ -937,6 +938,70 @@ <h3>Compiling Symbolic Circuits<a class="headerlink" href="#compiling-symbolic-c
 <p>Every <code class="xref py py-class docutils literal notranslate"><span class="pre">Backend</span></code> requires <code class="xref py py-class docutils literal notranslate"><span class="pre">NoSymbolsPredicate</span></code>, so it is necessary to instantiate all symbols before running a <code class="xref py py-class docutils literal notranslate"><span class="pre">Circuit</span></code>.</p>
 </div>
 </section>
+<section id="customised-passes">
+<h3>Customised Passes<a class="headerlink" href="#customised-passes" title="Permalink to this heading"></a></h3>
+<p>We have already seen that pytket allows users to combine passes in a desired order using <code class="xref py py-class docutils literal notranslate"><span class="pre">SequencePass</span></code>. An addtional feature is the <code class="xref py py-class docutils literal notranslate"><span class="pre">CustomPass</span></code> allowing users to define their own customised circuit transformation using pytket.
+The <code class="xref py py-class docutils literal notranslate"><span class="pre">CustomPass</span></code> class accepts a <cite>transform</cite> parameter.This <cite>transform</cite> is a python function that takes a <code class="xref py py-class docutils literal notranslate"><span class="pre">Circuit</span></code> as input and returns a <code class="xref py py-class docutils literal notranslate"><span class="pre">Circuit</span></code> as output.</p>
+<p>We will illustrate this using a function which simply replaces Pauli Z gates with a Pauli X and two Hadamard gates.</p>
+<p>We can now define our <code class="xref py py-class docutils literal notranslate"><span class="pre">CustomPass</span></code> using this function and apply it to a <code class="xref py py-class docutils literal notranslate"><span class="pre">Circuit</span></code> in the usual way.</p>
```

Comment:
A suggestion to neaten up the code snippet:
```from pytket import Circuit, OpType
from pytket.passes import CustomPass

def z_transform(input_circuit: Circuit) -> Circuit:
    circ_prime = Circuit(input_circuit.n_qubits) # Define a replacement circuit

    for cmd in input_circuit.get_commands(): # Iterate through all the gates in our input Circuit
        qubit_list = cmd.qubits # Qubit our gate is applied on (as a list)
        if cmd.op.type == OpType.Z: # If cmd is a Z gate, decompose to a H, X, H sequence.
            circ_prime.add_gate(OpType.H, qubit_list)
            circ_prime.add_gate(OpType.X, qubit_list)
            circ_prime.add_gate(OpType.H, qubit_list)
        else:
            # Otherwise apply the gate as usual
            circ_prime.add_gate(cmd.op.type, cmd.op.params, qubit_list)
    return circ_prime

DecomposeZPass = CustomPass(z_transform) # Define our pass

test_circ = Circuit(2) # Define a test Circuit for our pass
test_circ.Z(0)
test_circ.Z(1)
test_circ.CX(0, 1)
test_circ.Z(1)
test_circ.CRy(0.5, 0, 1)

DecomposeZPass.apply(test_circ) # Apply our pass to the test Circuit

test_circ.get_commands() # Commands of our transformed Circuit

Reply:
I'd also consider having the `z_transform` function and the construction of the `CustomPass` object as different code snippets.

Reply:
Do you know if this is possible in sphinx? I originally tried  to separate them but then it says `z_transfrom` is undefined. It seems that definitions of functions only apply within individual snippets/cells.

Reply:
I don't sorry - I guess if it isn't possible then it's fine as is, but worth having a little look through the docs just incase I think!